File: app/src/main/java/org/dhis2/usescases/main/program/ProgramContract.java
package org.dhis2.usescases.main.program;

import androidx.annotation.UiThread;

import org.dhis2.usescases.general.AbstractActivityContracts;
import org.hisp.dhis.android.core.organisationunit.OrganisationUnit;

import java.util.List;

import io.reactivex.functions.Consumer;

/**
 * QUADRAM. Created by ppajuelo on 18/10/2017.
 */
public class ProgramContract {

    interface View extends AbstractActivityContracts.View {

        Consumer<List<ProgramViewModel>> swapProgramModelData();

        @UiThread
        void renderError(String message);

        void openOrgUnitTreeSelector();

        void showHideFilter();

        void clearFilters();
    }

    public interface Presenter {
        void init(View view);

        void onItemClick(ProgramViewModel programModel);

        void showDescription(String description);

        List<OrganisationUnit> getOrgUnits();

        void dispose();

        void onSyncStatusClick(ProgramViewModel program);

        boolean areFiltersApplied();

        void showHideFilterClick();

        void clearFilterClick();
    }
}

File: app/src/main/java/org/dhis2/usescases/main/program/ProgramContract.kt
package org.dhis2.usescases.main.program

import androidx.annotation.UiThread
import io.reactivex.functions.Consumer
import org.dhis2.usescases.general.AbstractActivityContracts

/**
 * QUADRAM. Created by ppajuelo on 18/10/2017.
 */
class ProgramContract {

    interface View : AbstractActivityContracts.View {

        fun swapProgramModelData(): Consumer<List<ProgramViewModel>>

        fun showFilterProgress()

        @UiThread
        fun renderError(message: String)

        fun openOrgUnitTreeSelector()

        fun showHideFilter()

        fun clearFilters()
    }

    interface Presenter {
        fun init(view: View)

        fun onItemClick(programModel: ProgramViewModel)

        fun showDescription(description: String)

        fun dispose()

        fun onSyncStatusClick(program: ProgramViewModel)

        fun showHideFilterClick()

        fun clearFilterClick()
    }
}
